src/main/java/net/datafaker/idnumbers/NorwegianIdNumber.java

```diff
@@ -0,0 +1,126 @@
+    private static final int[] CHECKSUM_COEFFICIENTS_K2 = {5, 4, 3, 2, 7, 6, 5, 4, 3, 2};
+
+    @Override
+    public String countryCode() {
+        return "NO";
+    }
+
+    @Override
+    public PersonIdNumber generateValid(BaseProviders faker, IdNumberRequest request) {
+        LocalDate birthday = birthday(faker, request);
+        Gender gender = gender(faker, request);
+
+        String basePart = basePart(faker, birthday, gender);
+        String idNumber = "%s%02d".formatted(basePart, checksum(basePart));
+        return new PersonIdNumber(idNumber, birthday, randomGender(faker));
+    }
+
+    @Override
+    public String generateInvalid(BaseProviders faker) {
+        String valid = generateValid(faker);
+        String basePart = valid.substring(0, valid.length() - 2);
+        int invalidChecksum = (checksum(basePart) + 1) % 100;
+        return "%s%02d".formatted(basePart, invalidChecksum);
+    }
+
+    String basePart(BaseProviders faker, LocalDate birthday, Gender gender) {
+        String birthdayDigits = BIRTHDAY_FORMAT.format(birthday);
+
+        int sequenceNumber = CENTURIES
+            .stream()
+            .filter(e -> e.getKey().contains(birthday.getYear()))
+            .map(Map.Entry::getValue)
+            .map(interval -> faker.random().nextInt(interval.from, interval.to))
+            .findFirst()
+            .orElseThrow(() -> new IllegalArgumentException("Birthday is not in range of supported in Norway"));
+
+        int genderDigit = gender == Gender.FEMALE
+            ? faker.options().option(0, 2, 4, 6, 8)
+            : faker.options().option(1, 3, 5, 7, 9);
+
+        return "%s%02d%s".formatted(birthdayDigits, sequenceNumber, genderDigit);
+    }
+
+    /**
+     * k1 = 11 - ((3 × d1 + 7 × d2 + 6 × m1 + 1 × m2 + 8 × å1 + 9 × å2 + 4 × i1 + 5 × i2 + 2 × i3) mod 11),
+     * k2 = 11 - ((5 × d1 + 4 × d2 + 3 × m1 + 2 × m2 + 7 × å1 + 6 × å2 + 5 × i1 + 4 × i2 + 3 × i3 + 2 × k1) mod 11).
+     */
+    int checksum(String numbers) {
+        int k1 = modulo11(numbers, CHECKSUM_COEFFICIENTS_K1);
+        int k2 = modulo11(numbers + k1, CHECKSUM_COEFFICIENTS_K2);
+        return k1 * 10 + k2;
+    }
+
+    private int modulo11(String numbers, int[] checksumCoefficients) {
+        int checkSum = 0;
+        for (int i = 0; i < numbers.length(); i++) {
+            int digit = Character.getNumericValue(numbers.charAt(i));
+            checkSum += checksumCoefficients[i] * digit;
+        }
+
+        return (11 - checkSum % 11) % 10;
+    }
+
+    private record Interval(int from, int to) {
+        public static Interval of(String string) {
+            String[] split = string.split("-");
+            return new Interval(parseInt(split[0]), parseInt(split[1]));
+        }
+
+        public boolean contains(int value) {
+            return value >= from && value <= to;
+        }
+
+        public Interval reverse() {
+            return new Interval(to, from);
+        }
```

Comment:
these 2 seems to be leading to confusing results especially if we call like 
```
new Interval(3, 5)
contains(4)
reverse()
contains(4)
```

Reply:
probably better to have something in constructor like 
```java
public static Interval of(String string) {
  final int[] split = Arrays.stream("Asd".split("-")).mapToInt(Integer::parseInt).toArray();
  return new Interval(Math.min(split[0], split[1]), Math.max(split[0], split[1]));
}
```

Reply:
In the end, I've simplified all this map and removed using Intervals at all.